Gems/Terrain/Code/Source/TerrainRenderer/TerrainClipmapManager.cpp

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <TerrainRenderer/TerrainClipmapManager.h>
+#include <AzFramework/Terrain/TerrainDataRequestBus.h>
+#include <Atom/RPI.Public/Image/ImageSystemInterface.h>
+
+namespace Terrain
+{
+    namespace
+    {
+        [[maybe_unused]] static const char* TerrainClipmapManagerName = "TerrainClipmapManager";
+        static constexpr const char* MacroColorClipmapString = "MacroColorClipmaps";
+        static constexpr const char* MacroNormalClipmapString = "MacroNormalClipmaps";
+        static constexpr const char* DetailColorClipmapString = "DetailColorClipmaps";
+        static constexpr const char* DetailNormalClipmapString = "DetailNormalClipmaps";
+        static constexpr const char* DetailHeightClipmapString = "DetailHeightClipmaps";
+        static constexpr const char* DetailMiscClipmapString = "DetailMiscClipmaps";
```

Comment:
Is there a particular reason using name strings to lookup clipmaps? (I see your are using it for attachment image names too. 
Because everytime converting string to Name is an extra cost. 
I would suggest you use enum as index to look up the map.

And you can pass names only for creating attachment.